enhancements/sig-policy/66-delete-policy-resources/README.md

```diff
@@ -0,0 +1,237 @@
+
+### Goals
+
+1. Object deletion: objects that are created by policies will be deleted when
+the policy is deleted
+2. Opt-in: add an optional new field to the `ConfigurationPolicy` spec and default
+to leaving the object on the cluster upon policy deletion so that the behavior
+of old policies is not changed
+
+### Non-Goals
+
+1. Deleting objects not created by the parent policy: we will not support
+deleting objects that are not directly created by a `ConfigurationPolicy` when
+that `ConfigurationPolicy` is deleted (i.e. existing resources that are edited
+by the `ConfigurationPolicy`). We determined implementing this would create
+more problems than it would solve, such as users accidentally deleting objects
+that are created by the cluster or other policies.
+
+## Proposal
+
+We would add a new field to the `ConfigurationPolicy` spec called
+`spec.pruneObjectBehavior`. This field will be an enum with the options `DeleteAll`,
+`DeleteIfCreated`, and `None`. When this field is set to `DeleteAll`, the controller
+will automatically delete all objects referenced in the `ConfigurationPolicy` when
+the `ConfigurationPolicy` is deleted. For `DeleteIfCreated`, the controller
+will automatically delete any objects that it has created when the
+`ConfigurationPolicy` is deleted and leave the objects that existed already when the
+`ConfigurationPolicy` was created. When it is set to `None` or not set, the
+controller will leave any created objects intact once the `ConfigurationPolicy`
+is deleted. This will preserve existing behavior and make child object deletion
+“opt-in.”
+
+### User Stories
+
+As a k8s admin, I want resources that are created by a policy to be cleaned
+up automatically when I delete the policy that created them.
+
+### Implementation Details/Notes/Constraints 
+
+`pruneObjectBehavior` will be at the spec level for the `ConfigurationPolicy`
+and thus the value it is set to will apply to all object templates contained in
+the `ConfigurationPolicy`. This will make it simpler for users to set than having
+the field be set at the template level. If users would like some objects to
+remain on deletion and others to be deleted, they can separate those objects
+into multiple `ConfigurationPolicy` objects contained under the same parent policy.
+
+`ConfigurationPolicy` objects already maintain a list of objects they manage in their
+status field called `relatedObjects`. Since most of the information we would need
+to track which objects need to be deleted already exists in this part of the status,
+we propose adding a boolean field inside `status.relatedObjects.Object` that tells
+the controller whether a resource needs to be deleted when the `ConfigurationPolicy`
+is deleted. We would set this value when the config policy controller creates
+a resource during the handling of an “enforce” policy template. Additionally, we will
+add the object's UID to the status field so that we can determine whether the object
```

Comment:
It will be used to verify the identity of the object to be deleted. Say a config policy creates `pod/foo`. Then, between evaluation loops, something else deletes and recreates that pod. In this case, if the behavior is `DeleteIfCreated`, we don't want the policy to delete the pod. In some sense, the "responsibility" for the pod has been passed to whatever else recreated it.

With just the name, we can't detect this situation. With a UID, we can.

Reply:
Sounds good. @willkutler could you please add this to the doc?